src/libraries/System.Private.CoreLib/src/System/Resources/ResourceManager.cs

```diff
@@ -549,32 +549,33 @@ protected static CultureInfo GetNeutralResourcesLanguage(Assembly a)
             return ManifestBasedResourceGroveler.GetNeutralResourcesLanguage(a, out _);
         }
 
-        // IGNORES VERSION
         internal static bool IsDefaultType(string asmTypeName,
-                                           string typeName)
+                                           string defaultTypeName)
         {
             Debug.Assert(asmTypeName != null, "asmTypeName was unexpectedly null");
 
             // First, compare type names
-            int comma = asmTypeName.IndexOf(',');
-            if (((comma < 0) ? asmTypeName.Length : comma) != typeName.Length)
-                return false;
+            int firstComma = asmTypeName.IndexOf(',');
+            int typeNameLength = (firstComma != -1) ? firstComma : asmTypeName.Length;
```

Comment:
do we already have tests that exercise all these branches?

Reply:
The various branches would be typically taken for non-standard or ill-formed resource files. I do not see any test coverage for non-standard or ill-formed resource files, so this is a problem for every other branch in the resource parsing code. It may be nice to add this test coverage, but not in this PR.